### PR TITLE
NXP backend: Use new Neutron flow for failing constant pad tests.

### PR DESCRIPTION
**Pull request by MartinPavella**
### Summary
In Neutron SW SDK 3.1.0, some edge cases of the constant pad operator are no longer supported by the old flow. Affected tests now use the new flow where the pad is supported.

(Only the last commit is relevant)

### Test plan
Unit tests provided.

cc @robert-kalmar @JakeStevens @digantdesai

### CODE REVIEW

File: backends/nxp/tests/ir/converter/node_converter/test_constant_pad_nd_converter.py [hunk 1]
```diff
@@ -163,13 +163,12 @@ def test_constant_pad_nd__unsupported_paddings(input_shape, paddings, use_qat):
     assert any(node.name == "aten_constant_pad_nd_default" for node in nodes)
 
 
-@pytest.mark.xfail(reason="EIEX=855")
```
Comment (robert-kalmar) [inline backends/nxp/tests/ir/converter/node_converter/test_constant_pad_nd_converter.py:166]: Parametrize the test to it for legacy and new flow. Those two are distinct and the new mlir flow is experimental, hence the EIEX-855 is still valid. 

Yes, eventually the EIEX-855 will be closed as outdated, once the mlir flow is mature enough but we are not yet there. 
As we will do this for multiple operators define a `mlir_flow` as default parameter, like for qat - executorch/backends/nxp/tests/use_qat.py

Comment (MartinPavella) [reply backends/nxp/tests/ir/converter/node_converter/test_constant_pad_nd_converter.py:166]: Waiting for Ciprian's response to your email.

**Comment by MartinPavella** [comment]
@robert-kalmar  In Neutron SW 3.1.1, the new MLIR flow is the default. So can we merge this fix?

**Review by robert-kalmar — APPROVED** [review]
